api/hypershift/v1beta1/aws.go

```diff
@@ -60,6 +60,7 @@ type AWSNodePoolPlatform struct {
 }
 
 // PlacementOptions specifies the placement options for the EC2 instances.
+// +kubebuilder:validation:XValidation:rule="self.tenancy == 'host' ? !has(self.capacityReservation) : true", message="AWS Capacity Reservations cannot be used with Dedicated Hosts (tenancy 'host')"
```

Comment:
how about "dedicated"?

Reply:
capcityReservation + 'dedicated' tenancy is valid

> Capacity Reservations can't be used with Dedicated Hosts. Capacity Reservations can be used with Dedicated Instances.

from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-capacity-reservations.html#capacity-reservations-limitations